Scripts/Enemy/Enemy.cs

```diff
@@ -0,0 +1,102 @@
+using Godot;
+using System.Collections.Generic;
+
+public class Enemy : KinematicBody, IStatusHolder
+{
+    public Vector3 Velocity;
+    private readonly float gravity = 4.85f;
+    private float health = 100f;
+    private SpatialMaterial material;
+    private List<Status> statuses;
+
+    public override void _Ready()
+    {
+        material = new SpatialMaterial()
+        {
+            AlbedoColor = Colors.White,
+            FlagsTransparent = true
+        };
+
+        CSGCylinder cylinder = GetNode<CSGCylinder>("CSGCylinder");
+        cylinder.Material = material;
+        statuses = new List<Status>();

```

Comment:
Looking at the usage, it seems using a dictionary is more appropriate. Otherwise we need to iterate through the list to look for a status. O(n) worst case complexity.

Reply:
Lol that is true, did consider that but forgot to change